src/VirtualYAccessTrait.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Drupal\openy_gated_content;
+
+/**
+ * Trait Virtual Y Access Trait
+ *
+ * @package Drupal\openy_gated_content
+ */
+trait VirtualYAccessTrait {
+
+  /**
+   * Constant with virtual y editor role.
+   *
+   * @var string
+   */
+  public static $virtualYAccessEditorRole = 'virtual_ymca_editor';
```

Comment:
@anpolimus this still only allows a single `virtualYAccessEditorRole`. Would it be possible to make this an array which devs could extend/override? Some of the other check logic would have to be changed, but i think it's only 2 or 3 places.

Reply:
we can make it not a variable but a function

Reply:
@froboy, i've changed approach from variable to a method that returns an array

Reply:
Nice. I think that will work. I'm not able to test immediately, but visual review looks good.